client/app/components/conversation.coffee

```diff
@@ -42,29 +33,35 @@ _freezeHeader = ->
 module.exports = React.createClass
     displayName: 'Conversation'
 
-
     componentDidMount: ->
-        _getFullConversation.call @
-        _freezeHeader.call @
-        _scrollToActive.call @
+        unless @props.isConversationLoading
+            _freezeHeader.call @
+            _scrollToActive.call @
+
+
+    componentWillReceiveProps: (nextProps) ->
+        # If this conversation was removed
+        # redirect to messages list from mailbox
+        if nextProps.isConversationLoading and not nextProps.messages.length
+            setTimeout ->
```

Comment:
Just a note: I suppose setTimeout inside view are probably a dark pattern, and we may find amother way to handle let. Just a reminder, let's discuss it later ;)